bears/pug/PugLintBear.py

```diff
@@ -0,0 +1,240 @@
+import json
+import re
+
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.settings.Setting import typed_list
+
+from dependency_management.requirements.NpmRequirement import NpmRequirement
+
+
+@linter(executable='pug-lint',
+        use_stdout=True,
+        use_stderr=True)
+class PugLintBear:
+    """
+    A configurable linter and style checker for Pug (formerly Jade) that is a
+    clean, whitespace-sensitive template language for writing HTML.
+    """
+
+    LANGUAGES = {'Pug'}
+    REQUIREMENTS = {NpmRequirement('pug-lint', '2.4.0')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Syntax', 'Redundancy'}
+    SEE_MORE = 'https://github.com/pugjs/pug-lint'
+
+    _output_regex = re.compile(
+        r'(?P<line>\d+):?(?P<column>\d+)? (?P<message>.+)')
+
+    @staticmethod
+    def generate_config(filename, file,
+                        prohibit_block_expansion: bool=True,
+                        prohibit_class_attribute_with_static_value: bool=True,
+                        prohibit_class_literals_before_attributes: bool=True,
+                        prohibit_class_literals_before_id_literals: bool=True,
+                        prohibit_class_literals: bool=True,
+                        prohibit_duplicate_attributes: bool=True,
+                        prohibit_html_text: bool=True,
+                        prohibit_id_attribute_with_static_value: bool=True,
+                        prohibit_id_literals_before_attributes: bool=True,
+                        prohibit_id_literals: bool=True,
+                        prohibit_legacy_mixin_call: bool=True,
+                        prohibit_multiple_line_breaks: bool=None,
+                        prohibit_spaces_inside_attribute_brackets: bool=True,
+                        prohibit_string_interpolation: bool=True,
+                        prohibit_tag_interpolation: bool=True,
+                        prohibit_specific_attributes: typed_list(str)=None,
+                        prohibit_specific_tags: typed_list(str)=None,
+                        require_class_literals_before_attributes: bool=False,
+                        require_class_literals_before_id_literals: bool=False,
+                        require_id_literals_before_attributes: bool=False,
+                        require_lower_case_attributes: bool=True,
+                        require_lower_case_tags: bool=True,
+                        require_spaces_inside_attribute_brackets: bool=False,
+                        require_strict_equality_operators: bool=True,
+                        validate_div_tags: bool=True,
+                        validate_extensions: bool=True,
+                        validate_self_closing_tags: bool=True,
+                        preferred_quotation: str='\'',
+                        max_lines_per_file: int=None,
```

Comment:
`max_file_length` might refer to the number of chars inside a document, `max_lines_per_file` is less ambigious^^ So imo `max_lines_per_file` is better 👍